packages/abi-typegen/src/abi/types/StrSliceType.ts

```diff
@@ -14,10 +14,15 @@ export class StrSliceType extends AType implements IType {
   }
 
   public parseComponentsAttributes(_params: { types: IType[] }) {
+    const capitalizedName = 'StrSlice';
+
     this.attributes = {
-      inputLabel: 'StrSlice',
-      outputLabel: 'StrSlice',
+      inputLabel: capitalizedName,
+      outputLabel: capitalizedName,
     };
+
+    this.requiredFuelsMembersImports = [capitalizedName];
```

Comment:
We currently don't have a typegen test verifying that `StrSlice` is properly supported in the typegen _outputs_. I suggest you do the following:
1. Add a method in the [`full`](https://github.com/FuelLabs/fuels-ts/blob/master/packages/abi-typegen/test/fixtures/forc-projects/full/src/main.sw) contract which uses string slices
2. Run [this test](https://github.com/FuelLabs/fuels-ts/blob/cbc569ae3c7fea8b6405146f984b6c60861dd387/packages/abi-typegen/src/templates/contract/dts.test.ts#L17). Note that it'll be running both with and without the `--json-abi-with-callpaths` flag by using the `debug` and `release` flags as explained [here](https://github.com/FuelLabs/fuels-ts/pull/2372#discussion_r1610112456).
3. You should have a failing test because the generated file doesn't match the [template](https://github.com/FuelLabs/fuels-ts/blob/master/packages/abi-typegen/test/fixtures/templates/contract/dts.hbs). Update the template accordingly.
4. You might have to add support for `--json-abi-with-callpaths` for this type. You can look into other types that already support this flag (e.g. [`EvmAddressType`](https://github.com/FuelLabs/fuels-ts/blob/master/packages/abi-typegen/src/abi/types/EvmAddressType.ts#L10)).

Reply:
Thanks @nedsalk, will do. Will convert to draft for now. I'd initially flagged it when importing the typegen `str` factory in the `fuel-gauge` test.

Reply:
[Test added](https://github.com/FuelLabs/fuels-ts/pull/2685/commits/00ef33d86e2123c2b3b16aabb0b746bf6ca504a9), no changes needed for `--json-abi-with-callpaths`.